packages/deployment-server/test/deployment.spec.ts

```diff
@@ -235,3 +235,132 @@ test('Closes the i/o stream after a deployment request is cancelled', async (t)
     }
   })
 })
+
+test('returns an error when queue is full', async (t) => {
```

Comment:
A better, more focussed approach: 
1) Create device (util/factory.ts). Update status to unavailable. Deploy fails
2) Create device (util/factory.ts). Add MAX_QUEUE_SIZE requests, all work
3) Create device (util/factory.ts). Add MAX_QUEUE_SIZE+ 1 requests, last one fails